src/server/handlers/launchpad.js

```diff
@@ -97,11 +105,24 @@ export const setMemcached = (value) => {
   memcached = value;
 };
 
+class PreparedError extends Error {
+  constructor(status, body) {
+    super();
+    this.status = status;
+    this.body = body;
+  }
+}
+
 const responseError = (res, error) => {
-  if (error.response) {
+  if (error.status && error.body) {
+    // The error comes with a prepared representation.
+    return res.status(error.status).send(error.body);
+  } else if (error.response) {
     // if it's ResourceError from LP client at least for the moment
     // we just wrap the error we get from LP
-    return error.response.text().then(text => {
+    // XXX cjwatson 2016-12-22: Perhaps refactor to use the PreparedError
```

Comment:
Definitely worth refactoring at some point, although it's not clear how much we should push down to the LP client since ideally that should be split off to its own package and not hardcode BSI's error formats.  I'm going to defer this one.